include/experimental/__p0009_bits/layout_left.hpp

```diff
@@ -143,12 +143,14 @@ struct layout_left {
     mapping(OtherMapping const& other) // NOLINT(google-explicit-constructor)
       :__extents(other.extents())
     {
+       #ifndef __CUDA_ARCH__
        size_t stride = 1;
        for(size_type r=0; r<__extents.rank(); r++) {
          if(stride != other.stride(r))
            throw std::runtime_error("Assigning layout_stride to layout_left with invalid strides.");
```

Comment:
Any reason not to just do
```C++
assert(stride == other.stride(r) && "Assigning layout_stride to layout_left with invalid strides.");

Reply:
no

Reply:
actually:
```
Assertions should only be used to verify conditions that should be logically impossible to be false (read: sanity checks). These conditions should only be based on inputs generated by your own code. Any checks based on external inputs should use exceptions.
```

Reply:
https://docs.microsoft.com/en-us/cpp/cpp/errors-and-exception-handling-modern-cpp?view=msvc-170